### ISSUE TITLE
Rethinking the Tournaments

### ISSUE THREAD

**Issue by brianwgoldman** [siggame/colisee#4]
TL;DR - Triple Elim is overly complicated and limits how many places we can award. By combing a seeded single elimination tournament and full arena ELO calculation we can improve the final tournament and prize distribution.

**Proposed New System**

Once submissions close, use current arena ELO to construct a seeded single elimination tournament using only eligible teams. Care should be taken to obfuscate the bracket to hide the seed of each team. As with now, each match in this tournament is actually many games.

While that tournament is being viewed, the arena should be running at full blast in order to calculate final submission ELO. Care should be taken to reset ELO after seeding is complete. After waiting a sufficient amount of time, determine the final ranking of all teams based on post-submission ELO.

The largest prize (Gold) goes to the team that wins the single elimination tournament. Prizes in descending order of size are then awarded to the highest ranked student teams that didn't win the single elimination tournament (Silver, Bronze, Runner Up, Honorable Mention, etc).

**Advantages over Existing System**

Triple Elimination suffers from several problems:

1. Its complicated as hell to code.
2. Competitors may not understand how teams progress.
3. As we don't show the one- and two-loss brackets, observers just have to trust us on who deserves to advance out of them.
4. It only tells you the final ranking of 3 teams.
5. The final matches can be repetitive if there are two teams that only lose to each other.
6. While it has good guarantees on who gets awards, who gets what place can be noisy.

In comparison, the system I propose:

1. Can be implemented in perhaps 5 lines of pseudocode. 
2. Competitors are familiar with how single elimination tournaments work.
3. We show all matches needed to identify the Gold winner.
4. With the ELO results we can provide complete rankings of both student and non-student teams.
5. No two teams are shown playing each other twice
6. Good assurances that ranks are accurate, even if Gold may go to the "Cinderella Story" team.

**Comment by russleyshaw** [comment]
I've always through the purpose of the tournament is to determine (at least) 3 winners, for 1st, 2nd, and 3rd place. Triple elimination does exactly that. Just because previous implementations have been a bit wonkey maybe, doesn't mean it isn't the right choice. Single elimination only really gives us 1 winner, and we "infer" the others. Also if a competitor asks "how are the games played". I think its easier to point to triple elimination, rather than single elimination + ELO results (because most won't know what or how ELO is calculated. 

That's just my 2 cents, and I'd be curious to hear what other SIG-Game developers think.

**Comment by brianwgoldman** [comment]
> I've always through the purpose of the tournament is to determine (at least) 3 winners

I would argue the purpose of tournaments is three fold:

1. To be exciting to watch.
2. To be precise about how competitors are ranked.
3. To be comprehensible to competitors.

If we didn't have criteria 1, I would argue the best tournament structure would be Swiss, as it allows us to provide complete rankings with high precision, but there isn't a thrilling way to have everyone watch a Swiss tournament. Single Elimination is probably the most exciting tournament structure (every match is sudden death), and can ostensibly be used to award 4 places. The problem is that if you run the same tournament again (possibly with different seeding) you can get very different winners. In the past I've devised methods that maximize the first two, but they almost always end up being overly complicated. For instance, you could use Swiss to obtain the rankings, and then construct a single elimination tournament that produces the same top 4 ranks.

> Just because previous implementations have been a bit wonkey maybe, doesn't mean it isn't the right choice.

Its not just our implementation. To my knowledge there is no clean algorithm for correctly constructing arbitrarily large triple elimination brackets.

> I think its easier to point to triple elimination, rather than single elimination + ELO results (because most won't know what or how ELO is calculated.

During Saloon you all released the ELO table during competition. If you do so in the future, it will give competitors a constant measure how well they are doing overall. Giving them awards based on that ranking seems completely straight forward. In fact, I would argue that giving awards based *only* on table rank would maximize criteria 2 and 3, but it would fail 1. That is why I propose the combining it with single elimination.

> Single elimination only really gives us 1 winner, and we "infer" the others.

If you think that violates criteria 3, I'm fine making the single elimination prize separate from the ranking prize. Example:

- $120 for highest student ELO (First Place)
- $90 for second highest student ELO (Second Place)
- $60 for third highest student ELO (Third Place)
- (as many more like that as you want)
- $60 for winning the single elimination tournament (Highlander Award / Death Match Champion)

That way the "Place" you get is entirely controlled by ELO, but there is still the excitement of who gets the Highlander Award.

> Triple elimination does exactly that.

If you all are up for it, I would love to test how precise it really is. For instance, if you run triple elimination 20 times (with and without fixed seeding), do you get the same ordering of the top three? How does that compare with single elimination? Or my proposal? 

Its important to note that every tournament system has its own way of saying who is "best". Imagine a rock-paper-scissors tournament where you are the only paper, 80% of other teams are rock, and the rest are scissors. 

- In seeded Triple Elimination, there is a good chance you will not place since you'll likely face 3 scissors in a row.
- If you face a rock in your first match of Triple Elimination, you will likely win first place as all scissors lose to rocks before pairing with you.
- In round robin you will win first place as you beat 80% of opponents and second place only beats ~60%.

This is why I say tournaments should be precise (return the same result every time) instead of accurate (return results that are close to "right").

**Comment by Daniel17sep** [comment]
I like elimination because it gives a tournament bracket to show, but I also like ELO or Swiss tournaments because they _can_ be more fair and they _are_ faster to run.  I would not be against doing both like @brianwgoldman suggested.

**Comment by JacobFischer** [comment]
I basically agree with Goldman on everything.

I don't think it's written down anywhere exactly how our tournament should function, but I think an ideal tournament is easy to understand and fair. And, yes, I understand those are vague terms.

While I understand the appeal of Triple Elim, the amount of hand waving we have to do with the additional brackets is undesirable for a presentation. I know friends I've gotten to compete complain that we don't show how they lost because we skip over so much of the loser's bracket. Then we don't post results (which is another issue entirely), and it aggravates losers from coming back. For the record I'm not in favor of showing _more_ games. In fact I'd rather show _less_; because after 24 hours most people are burnt out.
